alc/backends/jack.cpp

```diff
@@ -508,15 +508,32 @@ std::string JackBackendFactory::probe(BackendType type)
         if(client == nullptr)
             throw al::backend_exception{ALC_INVALID_VALUE, "Failed to open client connection: 0x%02x",
                     status};
-        const char **ports{jack_get_ports(client, nullptr, nullptr, JackPortIsInput)};
+        const char **ports{jack_get_ports(client, nullptr, nullptr, JackPortIsPhysical|JackPortIsInput)};
         for (int i = 0; ports && ports[i] ; i++) {
+            const char* colon = strchr(ports[i], ':');
+            int len = static_cast<int>  (colon - ports[i]);
+            if (outnames.find(ports[i], 0, len) == std::string::npos) {
```

Comment:
This will break if there isn't a `:` in the port name (not likely to happen, but best to be defensive when dealing with external data). This will also be a problem if something like `alsa_out2` gets added before `alsa_out` (the latter will match against the former and not get added).

Reply:
>  This will break if there isn't a `:`

I just added a check if there isn't one : the complete name will be added

 >  alsa_out2 gets added before alsa_out

For this, including the final `:` by adding one to the length should be enough to force the device name. And with the list of custom names, we could remove the `:` afterward for more clarity.